src/XMLexport.cpp

```diff
@@ -981,452 +1033,419 @@ bool XMLexport::exportAction(QIODevice* device)
+//    writeStartElement("MudletPackage");
+//    writeAttribute(QStringLiteral("version"), mudlet::self()->scmMudletXmlDefaultVersion);
 
-    writeStartElement("KeyPackage");
-    bool isOk = writeKey(mpKey);
-    writeEndElement(); // </KeyPackage>
+//    writeStartElement("KeyPackage");
+//    bool isOk = writeKey(mpKey, pugi::xml_node());
+//    writeEndElement(); // </KeyPackage>
 
-    writeEndElement(); // </MudletPackage>
-    writeEndDocument();
+//    writeEndElement(); // </MudletPackage>
+//    writeEndDocument();
 
-    if (!isOk || hasError()) {
-        xmlBuffer.close();
-        return false;
-    }
+//    if (!isOk || hasError()) {
+//        xmlBuffer.close();
+//        return false;
+//    }
 
-    QClipboard* cb = QApplication::clipboard();
-    cb->setText(QString(xmlBuffer.buffer()), QClipboard::Clipboard);
+//    QClipboard* cb = QApplication::clipboard();
+//    cb->setText(QString(xmlBuffer.buffer()), QClipboard::Clipboard);
 
-    xmlBuffer.close();
-    return true;
+//    xmlBuffer.close();
+//    return true;
 }
 
-bool XMLexport::writeKey(TKey* pT)
+bool XMLexport::writeKey(TKey *pT, pugi::xml_node xmlParent)
 {
     bool isOk = true;
     if (!pT->mModuleMasterFolder && pT->exportItem) {
-        writeStartElement(pT->isFolder() ? "KeyGroup" : "Key");
+        auto keyContentsNode = xmlParent.append_child(pT->isFolder() ? "KeyGroup" : "Key");
+
+        // set the xml parent to the trigger we just made so nested triggers will get written to it
+        xmlParent = keyContentsNode;
 
-        writeAttribute("isActive", pT->shouldBeActive() ? "yes" : "no");
-        writeAttribute("isFolder", pT->isFolder() ? "yes" : "no");
+        keyContentsNode.append_attribute("isActive") = pT->shouldBeActive() ? "yes" : "no";
+        keyContentsNode.append_attribute("isFolder") = pT->isFolder() ? "yes" : "no";
 
         { // Blocked so that indentation reflects that of the XML file
-            writeTextElement("name", pT->mName);
-            writeTextElement("packageName", pT->mPackageName);
-            writeScriptElement(pT->mScript);
-            writeTextElement("command", pT->mCommand);
-            writeTextElement("keyCode", QString::number(pT->mKeyCode));
-            writeTextElement("keyModifier", QString::number(pT->mKeyModifier));
+            keyContentsNode.append_child("name").text().set(pT->mName.toUtf8().constData());
+            keyContentsNode.append_child("packageName").text().set(pT->mPackageName.toUtf8().constData());
+            writeScriptElement(pT->mScript, keyContentsNode);
+
+            keyContentsNode.append_child("command").text().set(pT->mCommand.toUtf8().constData());
+            keyContentsNode.append_child("keyCode").text().set(QString::number(pT->mKeyCode).toUtf8().constData());
+            keyContentsNode.append_child("keyModifier").text().set(QString::number(pT->mKeyModifier).toUtf8().constData());
         }
 
-        isOk = !hasError();
+        isOk = true;
```

Comment:
Not needed here now is it - as there is no possibility of errors so far in the method?